lib/fizzy/parser_expr.cpp

```diff
@@ -294,6 +326,8 @@ parser_result<Code> parse_expr(const uint8_t* pos, const uint8_t* end, bool have
         {
             if (frame.instruction != Instr::if_)
                 throw parser_error{"unexpected else instruction (if instruction missing)"};
+            frame.stack_height = 0;  // Reset stack height after if.
```

Comment:
Why is this reset?

Reply:
After passing `if` block, it reuses the same `frame` (`control_stack.top()`) for the `else` block - it has the same `instruction` and `immediates_offset`, but `stack_height` could have been changed in `if` block.

Reply:
Actually I think `unreachable` also has to be reset here?

Reply:
I think it should. In the Validation Algorithm it pops the control frame and then pushes new one with the same types.

Reply:
It would be nice to add a test with `br` inside `if`

Reply:
Fixed.

The `if-else` is poorly tested as there is big number of possible combinations.

Reply:
> It would be nice to add a test with `br` inside `if`

Added one with `unreachable`, but I will add one with `br` too. Also `br_table` is followed by unreachable...